tests/client_test.py

```diff
@@ -377,7 +411,178 @@ def test_intercept_unary_unary(self):
 
             mock_updater.assert_called_once_with(
                 mock_client_call_details, [mock_client_call_details.metadata[0],
-                interceptor.developer_token_meta,
-                interceptor.login_customer_id_meta])
+                                           interceptor.developer_token_meta,
+                                           interceptor.login_customer_id_meta])
 
             mock_continuation.assert_called_once()
+
+
+class LoggingInterceptorTest(TestCase):
+    """Tests for the google.ads.googleads.client.LoggingInterceptor class."""
+    _MOCK_CONFIG = {'test': True}
+    _MOCK_ENDPOINT = 'www.test-endpoint.com'
+    _MOCK_INITIAL_METADATA = [('developer-token', '123456'),
+                              ('header', 'value')]
+    _MOCK_CUSTOMER_ID = '123456'
+    _MOCK_REQUEST_ID = '654321xyz'
+    _MOCK_METHOD = 'test/method'
+    _MOCK_TRAILING_METADATA = (('request-id', _MOCK_REQUEST_ID),)
+    _MOCK_ERROR_MESSAGE = 'Test error message'
+
+    def _create_test_interceptor(self, config=_MOCK_CONFIG,
+                                 endpoint=_MOCK_ENDPOINT):
+        return google.ads.google_ads.client.LoggingInterceptor(config, endpoint)
+
+    def _get_mock_client_call_details(self):
+        mock_client_call_details = mock.Mock()
+        mock_client_call_details.method = self._MOCK_METHOD
+        mock_client_call_details.metadata = self._MOCK_INITIAL_METADATA
+        return mock_client_call_details
+
+    def _get_mock_request(self):
+        mock_request = mock.Mock()
+        mock_request.customer_id = self._MOCK_CUSTOMER_ID
+        return mock_request
+
+    def _get_trailing_metadata_fn(self, failed=False):
+        def mock_trailing_metadata_fn():
```

Comment:
In order to scope the inner function to the outer one, also for readability since this function is not used for any other purpose other than to be returned here.